### DISCUSSION TITLE
Move containers using mouse wheel

### DISCUSSION THREAD

**Discussion by ulugbek0217** [i3/i3#6275] [Q&A] [answered]
Hi there, I want to move a focused container to the next/prev workspace using Mod1+mouse wheel. The config is like this:
```
bindsym --whole-window --border Mod1+Shift+button5 move container to workspace next
bindsym --whole-window --border Mod1+Shift+button4 move container to workspace prev
```

It works well with existing workspaces. But I want to make like this: if the container in the latest workspace, it must be moved to a new workspace when I move it to next workspace. So if there are 3 workspaces and the container in the 3rd workspace, it must be moved to 4th workspace, not to 1st one. 
How can I do that?

**Comment by yusufaktepe** [comment, marked as answer]
This should work (_you need to install [jq](https://github.com/jqlang/jq)_):

```bash
#!/bin/bash

while read -r Used Focused; do
	[ "$Focused" = true ] && WSFocused=$Used
done < <(i3-msg -t get_workspaces | jq -r '.[] | [.num, .focused] | join(" ")')

next() {
	if [ "$WSFocused" = 9 ]; then
		i3-msg -q "move container to workspace 1"
	else
		i3-msg -q "move container to workspace $(( WSFocused + 1))"
	fi
}

prev() {
	if [ "$WSFocused" = 1 ]; then
		i3-msg -q "move container to workspace 9"
	else
		i3-msg -q "move container to workspace $(( WSFocused - 1))"
	fi
}

 case $1 in
 	prev) prev ;;
 	next|"") next ;;
 esac

```

Save this as `i3-mv` and make it executable; then you can bind it to your keys as:

```
bindsym --whole-window --border Mod1+Shift+button5 exec --no-startup-id i3-mv next
bindsym --whole-window --border Mod1+Shift+button4 exec --no-startup-id i3-mv prev
```

**Reply by ulugbek0217** [reply]
Thank you bro. This worked as a solution.
